src/test/scala/io/citrine/lolo/AccuracyTest.scala

```diff
@@ -0,0 +1,127 @@
+    val learner = new Bagger(baseLearner, numBags = nRow * nScal)
+    val error = computeMetrics(learner)
+    assert(error > noiseLevel, s"Can't do better than noise")
+    assert(error < 4.0, "Error increased, probably due to a change in configuration")
+  }
+
+  /**
+    * Check that a low-temperature Boltzmann Tree recovers the performance of a normal RF (with randomized pivots)
+    */
+  @Test
+  def testLowTLimit(): Unit = {
+    val errorStandardTree = {
+      val baseLearner = RegressionTreeLearner(
+        numFeatures = nFeat,
+        splitter = RegressionSplitter(randomizePivotLocation = true)
+      )
+      val learner = new Bagger(baseLearner, numBags = nRow * 16)
+      // println(s"Normal train time: ${Stopwatch.time(computeMetrics(learner))}")
+      computeMetrics(learner)
+    }
+    val errorAnnealingTree = {
+      val baseLearner = RegressionTreeLearner(
+        numFeatures = nFeat,
+        splitter = BoltzmannSplitter(temperature = 1.0e-9)
+      )
+      val learner = new Bagger(baseLearner, numBags = nRow * 16)
+      // println(s"Annealing train time: ${Stopwatch.time(computeMetrics(learner))}")
+      computeMetrics(learner)
+    }
+
+    val relativeDifference = 2.0 * Math.abs(errorAnnealingTree - errorStandardTree)/(errorAnnealingTree + errorStandardTree)
+    println(relativeDifference)
+    assert(relativeDifference < 0.01)
+  }
+}
+
+/**
+  * Driver code to study the performance vs temperature
+  */
+object AccuracyTest {
+
+  val trainingDataFull: Seq[(Vector[Any], Double)] = TestUtils.binTrainingData(TestUtils.generateTrainingData(2048, 48), inputBins = Seq((2, 32)))
+    .asInstanceOf[Seq[(Vector[Any], Double)]]
+
+  def computeMetrics(nRow: Int, nFeat: Int, nFeatSub: Int, nScal: Int, minInstances: Int, temperature: Double): (Double, Double) = {
+    val trainingData = trainingDataFull.take(nRow).map{case (f, l) => (f.take(nFeat), l)}
+    val splitter = if (temperature > 0) {
+      BoltzmannSplitter(temperature = temperature)
+    } else {
+      RegressionSplitter()
+    }
+    val baseLearner = new RegressionTreeLearner(
+      numFeatures = nFeatSub,
+      splitter = splitter,
+      minLeafInstances = minInstances
+    )
+    val learner = new Bagger(baseLearner, numBags = nRow * nScal, biasLearner = None)
+    val model = learner.train(trainingData).getModel()
+    val (features, labels) = trainingData.unzip
+    val predictions = model.transform(features)
+    val expected = predictions.getExpected().asInstanceOf[Seq[Double]]
+    val sigma = predictions.getUncertainty().get.asInstanceOf[Seq[Double]]
+    val pva: Seq[(Double, Double, Double)] = labels.indices.map{i =>
+      (labels(i), expected(i), sigma(i))
+    }
+    val rmse = Math.sqrt(pva.map{case (x, y, s) => Math.pow(x-y, 2)}.sum / pva.size)
+    val stdres = Math.sqrt(pva.map{case (x, y, s) => Math.pow((x-y) / s, 2)}.sum / pva.size)
+    (rmse, stdres)
+  }
+
+
+  def main(args: Array[String]): Unit = {
```

Comment:
Is this meant to be a test or is it just for diagnostics?

Reply:
If it's just diagnostic, can it be turned into a test? Is there any expected behavior we can check for?

Reply:
I'd like to defer maturing it to a follow-on PR.